subsys/event_manager_proxy/event_manager_proxy.c

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <spinlock.h>
+#include <event_manager.h>
+#include <event_manager_proxy.h>
+#include <logging/log.h>
+#include <ipc/ipc_service.h>
+
+LOG_MODULE_REGISTER(event_manager_proxy, CONFIG_EVENT_MANAGER_LOG_LEVEL);
+
+
+#define EMP_BIND_TIMEOUT K_MSEC(CONFIG_EVENT_MANAGER_PROXY_BIND_TIMEOUT_MS)
+
+/* Helpers - allow linker to get information about these structure sizes. */
+static struct event_type _emp_event_type_size_check
+	__used __attribute__((__section__("event_manager_proxy_event_type_size")));
+static struct event_type *_emp_event_type_pointer_size_check
+	__used __attribute__((__section__("event_manager_proxy_event_type_pointer_size")));
+
+/* Array used for inter-core event type mapping. */
+extern struct event_type *event_manager_proxy_array[];
+extern struct event_type *_event_manager_proxy_array_list_end[];
+
+
+/** @brief Command codes used by the proxy. */
+enum emp_cmd_code {
+	/** Register for an event on another core. */
+	EMP_CMD_REGISTER,
+
+	/** Start event transmission. No commands after this point, only events. */
+	EMP_CMD_START,
+
+	/** Number of commands. */
+	EMP_CMD_COUNT
+};
+
+/**
+ * @brief The command base structure.
+ */
+struct emp_cmd {
+	/** The command code */
+	enum emp_cmd_code code;
+};
+
+/**
+ * @brief The command structure used to register.
+ */
+struct emp_cmd_register {
+	/** The command code. */
+	enum emp_cmd_code code;
+
+	/** The event identifier to be used for the matching name. */
+	const struct event_type *id;
+
+	/** The name of the event. */
+	char name[];
+};
+
+/** @brief Inter-core communication data. */
+struct emp_ipc_data {
+	/** IPC ndpoint. */
+	struct ipc_ept ept;
+
+	/** Endpoint configuration - local copy to allow setting @ref ipc_ept_cfg::priv. */
+	struct ipc_ept_cfg ept_cfg;
+
+	/** Structure is used. */
+	bool used;
```

Comment:
Right. Seems like leftover.

Reply:
It seems that removing this flag would drive us into some races conditions. From the event_manager_proxy_on_event_process function we should configure ::event_type_map set before registering endpoint. But from the handle_remote_command point of view it should be set after registering endpoint. This can be solved by adding k_simlock, but then we would replace simple flag by simlock structure. I am not sure if this is the price worth to pay.